src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -140,6 +140,7 @@ private sealed class RunningInstance
 
             // Calling RunLoopStop multiple times SegFaults so protect the call to it
             private bool _stopping;
+            private object _stopLock = new object();
```

Comment:
Rather than allocating a new object, you could lock on some other non-exposed object you already have, e.g.
```C#
private object StopLock => this;
```
Alternatively, if you want to allocate and use a dedicated object, it'd be good to make it readonly.